src/redraw.mm

```diff
@@ -70,123 +70,65 @@ - (void) doAction:(RedrawCode::Enum)code withItem:(const object &)item_o
 {
     int item_sz = item_o.via.array.size;
     object *arglists = item_o.via.array.ptr + 1;
-    int narglists = item_sz - 1;
-
 
     if (code == RedrawCode::put) {
+        // Separate Front- and Backend colors
+        NSColor *fg = [mTextAttrs objectForKey:NSForegroundColorAttributeName];
+        NSColor *bg = [mTextAttrs objectForKey:NSBackgroundColorAttributeName];
 
-        /* This loop is getting a bit messy; feel free to rewrite. But, these
-           lines should render properly and be aligned:
-
-            iiiiiiiiiiiiiiiiiiii|
-            MMMMMMMMMMMMMMMMMMMM|
-            サパラサパサパラサパ|
-            བཔོལག་བཔོལག་བཔོལག་བཔོལག་|
+        // Remove Background color from text attributes
+        NSMutableDictionary *textAttrs = [[[NSMutableDictionary alloc] init] autorelease];
+        [textAttrs setDictionary:mTextAttrs];
+        [textAttrs removeObjectForKey:NSBackgroundColorAttributeName];
 
-            If we start drawing more than one char at a time, OSX starts
-            mashing the chars together in some scripts. Drawing char by char
-            is slower though, so not ideal.
-        */
-
-        static std::vector<std::string> runs;
-        runs.clear();
+        // Reverse bg and fg color if necessary
+        if (mReverseVideo) {
+            [textAttrs setObject:bg forKey:NSForegroundColorAttributeName];
+            bg = fg;
+        }
 
-        static std::vector<int> lens;
-        lens.clear();
+        // Draw background separately [width = item_sz]
+        NSRect bgrect = [self
+            viewRectFromCellRect:CGRectMake(mCursorPos.x, mCursorPos.y, item_sz - 1, 1)];
+        [bg set];
+        NSRectFill(bgrect);
 
-        static std::string run;
-        run.clear();
+        // Init string
+        NSString *nsrun = @"\u202d";
```

Comment:
Please add a comment explaining the meaning of \u202d. (It used to be near the `// Force left-to-right rendering` comment)